### ISSUE TITLE
Error when iframe is called without a block

### ISSUE THREAD

**Issue by mdesantis** [natritmeyer/site_prism#227]
Hello! Thank you for this wonderful library!

I have a page with a payments form, with a radio button. When the user clicks the radio button, some iframes with the credit card details forms gets added to the page. I should fill the credit card number field within the iframe that gets loaded. This is the test code:

```ruby
page = Checkout::PaymentPage.new
page.payment_form.credit_card_payment_method_radio_button.click # loads the iframe
page.wait_for_credit_card_number_iframe
expect(page).to have_credit_card_number_iframe # passes
page.credit_card_number_iframe #raises an error
# Here I should call `page.credit_card_number_iframe` in order to set the credit card number
```

Which is causing the following error:

```
NoMethodError:
  undefined method `call' for nil:NilClass
  Did you mean?  caller
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/site_prism-2.11/lib/site_prism/element_container.rb:60:in `block (2 levels) in iframe'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/capybara-2.18.0/lib/capybara/session.rb:426:in `within_frame'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/capybara-2.18.0/lib/capybara/dsl.rb:50:in `block (2 levels) in <module:DSL>'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/site_prism-2.11/lib/site_prism/element_container.rb:59:in `block in iframe'
# ./spec/features/purchase_spec.rb:113:in `block (2 levels) in <top (required)>'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:254:in `instance_exec'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:254:in `block in run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:500:in `block in with_around_and_singleton_context_hooks'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:457:in `block in with_around_example_hooks'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:466:in `block in run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:606:in `block in run_around_example_hooks_for'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:342:in `call'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-rails-3.7.2/lib/rspec/rails/adapters.rb:127:in `block (2 levels) in <module:MinitestLifecycleAdapter>'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:447:in `instance_exec'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:447:in `instance_exec'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:375:in `execute_with'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:608:in `block (2 levels) in run_around_example_hooks_for'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:342:in `call'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:609:in `run_around_example_hooks_for'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/hooks.rb:466:in `run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:457:in `with_around_example_hooks'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:500:in `with_around_and_singleton_context_hooks'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example.rb:251:in `run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example_group.rb:628:in `block in run_examples'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example_group.rb:624:in `map'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example_group.rb:624:in `run_examples'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example_group.rb:590:in `run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `block (3 levels) in run_specs'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `map'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `block (2 levels) in run_specs'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1926:in `with_suite_hooks'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:113:in `block in run_specs'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:79:in `report'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:112:in `run_specs'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:87:in `run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:71:in `run'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:45:in `invoke'
# /home/user/.asdf/installs/ruby/2.3.6/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.1/exe/rspec:4:in `<top (required)>'
# /home/user/.asdf/installs/ruby/2.3.6/bin/rspec:23:in `load'
# /home/user/.asdf/installs/ruby/2.3.6/bin/rspec:23:in `<main>'
```

## Environment
- Operating System: Ubuntu
- Gems / Versions: site_prism 2.11, capybara 2.18.0, ruby 2.3.6
- Test Framework: RSpec
- Browser/s Driver/s being used: Firefox 59.0.1, Geckodriver 0.20.0

**Comment by mdesantis** [comment]
Ok, the error was mine: [site_prism iframe element method needs a block in order to interact with](https://github.com/natritmeyer/site_prism#interacting-with-an-iframes-contents). Anyway I think site_prism could raise a friendly error message if the iframe method gets called without a block, what do you think?

**Comment by luke-hill** [comment]
I was actually having a discussion about this the other week. The short answer is no we shouldn't. The longer answer is yes potentially.

Reasoning why: We wrap capybara's behaviour. I will leave this open to revisit sometime in the next week or so and have another think about this. I don't think what you're suggesting is a huge amount of work. Just whether we should do it here.

Thanks for the Issue Report and Self-Triaging.

**Comment by mdesantis** [comment]
Ok, I'm gonna change the issue subject, since there isn't any issue with iframes loaded after javascript event

**Comment by luke-hill** [comment]
@mdesantis From your earlier message (About willing to make a PR if I'd accept it), the answer is Gladly. Feel free to raise a PR to update docs / messages / anything else you think would help.

**Comment by luke-hill** [comment]
Ping @twalpole What are your thoughts on "where" this error should live?

**Comment by luke-hill** [comment]
Cheers. Yeh we check in various places for it. We'll add a check into our API, when I get round to it.

**Comment by luke-hill** [comment]
Closed by 70e3b1dbc50b5bee28fb2e763f27d3332b01b730 - Released in v2.15
